src/test/resources/org/rust/lang/core/parser/fixtures/complete/diesel_macros.rs

```diff
@@ -0,0 +1,712 @@
+                    ::join_clause(&$through::table, join_type);
+                parent_to_through.combine_with(through_to_child)
+            }
+        }
+    }
+}
+
+/// Takes a query `QueryFragment` expression as an argument and returns a string
+/// of SQL with placeholders for the dynamic values.
+///
+/// # Example
+///
+/// ### Returning SQL from a count statment:
+///
+/// ```rust
+/// # #[macro_use] extern crate diesel;
+/// # table! {
+/// #     users {
+/// #         id -> Timestamp,
+/// #         n -> Integer,
+/// #     }
+/// # }
+/// # // example requires setup for users table
+/// # use self::users::dsl::*;
+/// # use diesel::*;
+/// #
+/// # fn main() {
+/// let sql = debug_sql!(users.count());
+/// assert_eq!(sql, "SELECT COUNT(*) FROM `users`");
+/// # }
+/// ```
+#[macro_export]
+macro_rules! debug_sql {
+    ($query:expr) => {{
+        use $crate::query_builder::{QueryFragment, QueryBuilder};
+        use $crate::query_builder::debug::DebugQueryBuilder;
+        let mut query_builder = DebugQueryBuilder::new();
+        QueryFragment::<$crate::backend::Debug>::to_sql(&$query, &mut query_builder).unwrap();
+        query_builder.finish()
+    }};
+}
+
+/// Takes takes a query `QueryFragment` expression as an argument and prints out
+/// the SQL with placeholders for the dynamic values.
+///
+/// # Example
+///
+/// ### Printing SQL from a count statment:
+///
+/// ```rust
+/// # #[macro_use] extern crate diesel;
+/// # table! {
+/// #     users {
+/// #         id -> Timestamp,
+/// #         n -> Integer,
+/// #     }
+/// # }
+/// # // example requires setup for users table
+/// # use self::users::dsl::*;
+/// # use diesel::*;
+/// #
+/// # fn main() {
+/// print_sql!(users.count());
+/// # }
+/// ```
+#[macro_export]
+macro_rules! print_sql {
+    ($query:expr) => {
+        println!("{}", &debug_sql!($query));
+    };
+}
+
+// The order of these modules is important (at least for those which have tests).
+// Utililty macros which don't call any others need to come first.
+#[macro_use] mod internal;
+#[macro_use] mod parse;
+#[macro_use] mod query_id;
+#[macro_use] mod static_cond;
+#[macro_use] mod macros_from_codegen;
+
+#[macro_use] mod as_changeset;
+#[macro_use] mod associations;
+#[macro_use] mod identifiable;
+#[macro_use] mod insertable;
+
+#[cfg(test)]
+mod tests {
+    use prelude::*;
+
+    table! {
+        foo.bars {
+            id -> Integer,
+            baz -> Text,
+        }
+    }
+
+    mod my_types {
+        pub struct MyCustomType;
+    }
+
+    table! {
+        use types::*;
+        use macros::tests::my_types::*;
+
+        table_with_custom_types {
+            id -> Integer,
+            my_type -> MyCustomType,
+        }
+    }
+
+    #[test]
+    fn table_with_custom_schema() {
+        let expected_sql = "SELECT `foo`.`bars`.`baz` FROM `foo`.`bars`";
+        assert_eq!(expected_sql, debug_sql!(bars::table.select(bars::baz)));
+    }
+}
```

Comment:
Let's remove comments, tests and everything else which is not a `macro_rules` definition. 

These parser tests are a bit annoying in that if you tweak parser or PSI structure, they tend to get invalidated as collateral damage, even if the changes were done to an unrelated part of the grammar. So it makes sense to keep them short and dense!